src/odb/src/db/dbPGpin.cpp

```diff
@@ -0,0 +1,222 @@
+    dbBPin * r_bpin = dbBPin::create(r_bterm);
+    r_bpin->setPlacementStatus("FIRM");
+
+    int direction = pdn_wire->getDir();
+    if (direction == 1) {
+      // for horizontal
+      if (position == "left") {
+        int dx = r2grid((pdn_wire->xMax() - pdn_wire->xMin())/num_connection_points/2, tech);
+        xMin[n] = pdn_wire->xMin() + n*dx;
+        xMax[n] = xMin[n] + dx;
+
+
+      } else if (position == "right") {
+        int dx = r2grid((pdn_wire->xMax() - pdn_wire->xMin())/num_connection_points/2, tech);
+        xMin[n] = pdn_wire->xMin() + (pdn_wire->xMax() - pdn_wire->xMin())/2 + n*dx;
+        xMax[n] = xMin[n] + dx;
+
+
+      } else if (position == "middle") {
+        int dx = r2grid((pdn_wire->xMax() - pdn_wire->xMin())/num_connection_points/2, tech);
+        xMin[n] = pdn_wire->xMin() + (pdn_wire->xMax() - pdn_wire->xMin())/4 + n*dx;
+        xMax[n] = xMin[n] + dx;
+
+      } else if (position == "default") {
+        int dx = r2grid((pdn_wire->xMax() - pdn_wire->xMin())/num_connection_points, tech);
+        xMin[n] = pdn_wire->xMin() + n*dx;
+        xMax[n] = xMin[n] + dx;
+
+      } else {
+        std::cout << "WARNING.";
+        break;
+      }
+      dbBox::create(r_bpin, pdn_wire->getTechLayer(), xMin[n], pdn_wire->yMin(), xMax[n], pdn_wire->yMax()); //create physical box for net
+    } 
+    else if (direction == 0) {
+      //for vertical
+      // for horizontal
+      if (position == "left") {
+        int dy = r2grid((pdn_wire->yMax() - pdn_wire->yMin())/num_connection_points/2, tech);
+        yMin[n] = pdn_wire->yMin() + n*dy;
+        yMax[n] = yMin[n] + dy;
+
+      } else if (position == "right") {
+        int dy = r2grid((pdn_wire->yMax() - pdn_wire->yMin())/num_connection_points/2, tech);
+        yMin[n] = pdn_wire->yMin() + (pdn_wire->yMax() - pdn_wire->yMin())/2 + n*dy;
+        yMax[n] = yMin[n] + dy;
+
+      } else if (position == "middle") {
+        int dy = r2grid((pdn_wire->yMax() - pdn_wire->yMin())/num_connection_points/2, tech);
+        yMin[n] = pdn_wire->yMin() + (pdn_wire->yMax() - pdn_wire->yMin())/4 + n*dy;
+        yMax[n] = yMin[n] + dy;
+
+      } else if (position == "default") {
+        int dy = r2grid((pdn_wire->yMax() - pdn_wire->yMin())/num_connection_points, tech);
+        yMin[n] = pdn_wire->yMin() + n*dy;
+        yMax[n] = yMin[n] + dy;
+
+      } else {
+        std::cout << "WARNING.";
+        break;
+      }
+      dbBox::create(r_bpin, pdn_wire->getTechLayer(), pdn_wire->xMin(), yMin[n], pdn_wire->xMax(), yMax[n]); //create physical box for net
+
+    } else {
+      std::cout << "WARNING.";
+    }
+  }
+  
+}
+
+void dbPGpin::create_custom_connections (dbBlock* block, const char* nett, const char* instt, const char* itermm) {
+  dbNet* net = block->findNet(nett);
+
+  dbInst* inst = block->findInst(instt);
+  dbITerm* iterm = inst->findITerm(itermm);
+  iterm->connect(net);
+
+}
```

Comment:
This has nothing to do with the dbPGpin object.  Why is it a method?  Why all the duplicate letters in nett, instt, termm?